### PR TITLE
Vehicle control additions & fixes

### PR DESCRIPTION
**Pull request by Tayipc38**
Ill draft my current changes and take any suggestion into consideration.

Changes:
- [X] Improve the method of playing animations
- [X] Convert UI elements to components to make use of in-built fiber pool usage.
- [X] Remove summon distance (Nobody seemed to need it lol).
- [X] Allow vehicles other than cars.
- [x] Introduce a method of manual controlled vehicle selection #1876.
- [x]  Optional vehicle information.
- [x] Patch bugs.
- [x] Other suggestions...

### CODE REVIEW

**Comment by GTMoraes** [comment]
I'm using the changes on my own fork.

Few bugs I noticed so far:
- Selecting to control the Closest Vehicle and then choosing to control the Personal Vehicle, without a spawned personal vehicle, crashes the game.
- When selected to control the Personal Vehicle, the checkmark to turn engine on/off is bugged. It's always unchecked and doesn't change according to status. Also, it only turns the engine on, when the "Use Animation" checkbox is marked, and only turns the engine off when the same checkbox is unmarked.
Using the "Last Driven", it behaves correctly, although it doesn't turn the engine on or off, when the Use Animations is checked off (which was the behavior from before this update)

I'm still trying it out so far, but looks promising! Thanks for considering my suggestion to control other vehicles than the previous one.

**Comment by Tayipc38** [comment]
> I'm using the changes on my own fork.
> 
> Few bugs I noticed so far:
> 
> * Selecting to control the Closest Vehicle and then choosing to control the Personal Vehicle, without a spawned personal vehicle, crashes the game.
> * When selected to control the Personal Vehicle, the checkmark to turn engine on/off is bugged. It's always unchecked and doesn't change according to status. Also, it only turns the engine on, when the "Use Animation" checkbox is marked, and only turns the engine off when the same checkbox is unmarked.
>   Using the "Last Driven", it behaves correctly, although it doesn't turn the engine on or off, when the Use Animations is checked off (which was the behavior from before this update)
> 
> I'm still trying it out so far, but looks promising! Thanks for considering my suggestion to control other vehicles than the previous one.

Thanks for the heads up. This definitely needs more before it can be considered remotely done.

**Comment by Yimura** [comment]
@DayibBaba stale PR

**Comment by Tayipc38** [comment]
> @DayibBaba stale PR

Not planning on working on this for the foreseeable future.

EDIT: The future is now old man

**Comment by GTMoraes** [comment]
This PR as is, works reasonably OK.

The bugs I mentioned before are still present, and sometimes, switching from "Personal" to "Last driven" crashes the game when something that I can't really put my finger at, happens with the last driven vehicle. 
It's a rare event, but when it happens, it's after a mission and the vehicle despawns/disappears. I can read that the last vehicle model is a NULL and the game immediately crashes. Most of the time, though, even though the mission vehicle disappeared or whatever, it just displays the "last vehicle" as if it still existed, but does nothing. Next vehicle you enter, it works fine.

Last time I recall that it happened, was when I had to fly a police helicopter to take pictures of someone in a balcony. After the mission ended, I landed the chopper somewhere and when I opened the vehicle menu, it listed the last vehicle as NULL, then the game froze and crashed.   
But it also happened with normal vehicles.

Probably not ready for prime time, but looks close.

This is, easily, my most used feature from the whole menu. 99% of the time it works fine, and the known bugs I listed before are workaroundable.

**Comment by Tayipc38** [comment]
> This PR as is, works reasonably OK.
> 
> The bugs I mentioned before are still present, and sometimes, switching from "Personal" to "Last driven" crashes the game when something that I can't really put my finger at, happens with the last driven vehicle. 


Happens due to the personal vehicle not existing in the script. There is no check whether one exists before switching to it. The handle is fetched from a global and casted into a Vehicle. I have this fixed on my local but have recently deleted gta to reserve space for Starfield.

File: src/services/vehicle/vehicle_control_service.cpp [hunk 1]
```diff
@@ -399,13 +362,42 @@ namespace big
...
 		keep_controlled_vehicle_data_updated(m_controlled_vehicle);
+
+		if (!g.window.vehicle_control.opened)
+			return;
+
+		switch (m_selection_mode)
+		{
+		case eControlledVehSelectionMode::LAST_DRIVEN: get_last_driven_vehicle(); break;
+		case eControlledVehSelectionMode::PERSONAL: get_personal_vehicle(); break;
+		case eControlledVehSelectionMode::CLOSEST: get_closest_vehicle(); break;
```
Comment (Yimura) [inline src/services/vehicle/vehicle_control_service.cpp:400]: case, function and break on one line?

**Review by Yimura — CHANGES_REQUESTED** [review]
Waiting for PR on translation repo.

File: src/services/vehicle/vehicle_control_service.cpp (hunk 1 above)
Comment (Tayipc38) [reply src/services/vehicle/vehicle_control_service.cpp:400]: The format shortcut did this, not me.

**Review by Yimura — APPROVED** [review]
Should be ready to merge now, will handle the other PR's that still need their translations done.
Then after merging the translations this will go into master.

**Pull request merged**